scripts/common/leaderboard.ts

```diff
@@ -32,13 +32,36 @@ export enum LeaderboardStatusType {
 	UNAUTHORIZED_FRIENDS_LIST = 6
 }
 
-export function getMainTrack(mapData: MMap, gamemode: Gamemode): Leaderboard {
+export function getTrack(
+	mapData: MMap,
+	gamemode: Gamemode,
+	trackType: TrackType = TrackType.MAIN,
+	trackNum: number = 1,
+	style: number = 0
+): Leaderboard {
 	return mapData.leaderboards.find(
 		(leaderboard) =>
-			leaderboard.gamemode === gamemode && leaderboard.trackType === TrackType.MAIN && leaderboard.style === 0
+			leaderboard.gamemode === gamemode &&
+			leaderboard.trackType === trackType &&
+			leaderboard.trackNum === trackNum &&
+			leaderboard.style === style
 	);
 }
 
 export function getNumZones(mapData: MMap): number {
 	return mapData.leaderboards.filter((leaderboard) => leaderboard.trackType === TrackType.STAGE).length;
 }
+
+export function getNumBonuses(mapData: MMap): number {
+	return mapData.leaderboards.filter((leaderboard) => leaderboard.trackType === TrackType.STAGE).length;
```

Comment:
This should be `TrackType.BONUS`

Reply:
Yeah this is fixed already, just haven't pushed yet. I'd hold off on reviewing today, I don't have the energy to get this branch finalized tonight and there's a lot missing.

Also note that this function is wrong in that it can count multiple gamemodes on the same track, same issue with C++.

Reply:
Fixed now